lib/backend/k8s/k8s.go

```diff
@@ -86,7 +104,89 @@ func NewKubeClient(kc *KubeConfig) (*KubeClient, error) {
 		return nil, err
 	}
 	log.Debugf("Created k8s clientSet: %+v", cs)
-	return &KubeClient{clientSet: cs}, nil
+
+	tprClient, err := buildTPRClient(config)
+	if err != nil {
+		return nil, err
+	}
+	kubeClient := &KubeClient{
+		clientSet: cs,
+		tprClient: tprClient,
+	}
+
+	// Ensure the necessary ThirdPartyResources exist in the API.
+	err = kubeClient.createThirdPartyResources()
+	if err != nil {
+		// Couldn't set up resources.  Start a goroutine to retry.
+		go kubeClient.ensureThirdPartyResources()
+	}
+
+	return kubeClient, nil
+}
+
+// ensureThirdPartyResources ensures the necessary thirdparty resources are created.
+// This runs as a goroutine, and will retry periodically until success.
+func (c *KubeClient) ensureThirdPartyResources() {
+	for err := c.createThirdPartyResources(); err != nil; {
+		log.Warnf("Failed to ensure necessary third party resources, will retry: %s", err)
+		time.Sleep(5 * time.Second)
+	}
+}
+
+// createThirdPartyResources creates the necessary third party resources if they
+// do not already exist.
+func (c *KubeClient) createThirdPartyResources() error {
+	// Ensure a resource exists for Calico global configuration.
+	tpr := extensions.ThirdPartyResource{
+		ObjectMeta: kapiv1.ObjectMeta{
+			Name:      "global-config.projectcalico.org",
+			Namespace: "kube-system",
+		},
+		Description: "Calico Global Configuration",
+		Versions:    []extensions.APIVersion{{Name: "v1"}},
+	}
+	_, err := c.clientSet.Extensions().ThirdPartyResources().Create(&tpr)
+	if err != nil {
+		// Don't care if it already exists.
+		if !kerrors.IsAlreadyExists(err) {
+			return goerrors.New(fmt.Sprintf("failed to create ThirdPartyResource %s: %s", tpr.ObjectMeta.Name, err))
+		}
+	}
```

Comment:
I still need to figure out what to do here.

A watch on global configs fails until an instance of the TPR is created.  I tried to create an instance here at startup, but it's not only yucky it seems to not work. idk if the requests are too fast, but adding a 5 second sleep between creating the TPR and the GlobalConfig makes it work.  Retries don't seem to help...

We should talk tomorrow - this needs to either go in or be punted until v2.1.

If it's punted, there are two key bugfixes in this PR that need to go into v2.0